provisioning/transport/mqtt/src/ProvisioningChannelHandlerAdapter.cs

```diff
@@ -309,7 +322,25 @@ private async Task VerifyPublishPacketTopicAsync(IChannelHandlerContext context,
             {
                 if (Enum.TryParse(match.Groups[1].Value, out HttpStatusCode statusCode))
                 {
-                    if (statusCode >= HttpStatusCode.BadRequest)
+                    if (statusCode == HttpStatusCode.BadRequest)
+                    {
+                        await FailWithExceptionAsync(
+                         context,
+                         new ProvisioningTransportException(
+                             jsonData,
+                             null,
+                             false)).ConfigureAwait(false);
+                    }
+                    else if ((int)statusCode == 429)
```

Comment:
>(int)statusCode == 429) [](start = 29, length = 23)

looks like this condition is covered in line 347 below.

Reply:
As per your note [here](https://github.com/Azure/azure-iot-sdk-csharp/pull/750#discussion_r249187971) aren't 429 and 400 special cases where we cannot parse into a ProvisioningErrorDetails object? So don't we need these special cases where we do not parse into a json object?

Reply:
Sorry, I was answering the question of whether the response would contain the status of 'Assigning'.  Anush is correct and you should receive an object of type 'ProvisioningServiceErrorDetails' which you are deserializing on line 345.

---
In reply to: [255747413](https://github.com/Azure/azure-iot-sdk-csharp/pull/791#discussion_r255747413) [](ancestors = 255747413)

Reply:
Ah okay, I'll leave the error parsing as it was, then.